tms/starter/starter.py

```diff
@@ -0,0 +1,211 @@
+    taskforce_uuid: str,
+    # condor args
+    do_transfer_output: bool,
+    max_worker_runtime: int,
+    n_cores: int,
+    priority: int,
+    worker_disk_bytes: int,
+    worker_memory_bytes: int,
+) -> dict[str, Any]:
+    """Make the condor job description (dict)."""
+
+    # NOTE:
+    # In the newest version of condor we could use:
+    #   universe = container
+    #   container_image = ...
+    #   arguments = python -m ...
+    # But for now, we're stuck with:
+    #   executable = ...
+    #   +SingularityImage = ...
+    #   arguments = /usr/local/icetray/env-shell.sh python -m ...
+    # Because "this universe doesn't know how to do the
+    #   entrypoint, and loading the icetray env file
+    #   directly from cvmfs messes up the paths" -DS
+
+    environment_str = " ".join(f"{k}={v}" for k, v in environment.items())
+    input_files_str = " ".join(input_files)
+
+    # write
+    submit_dict = {
+        "executable": "/bin/bash",
+        "arguments": arguments,
+        "+SingularityImage": f'"{image}"',  # must be quoted
+        "Requirements": "HAS_CVMFS_icecube_opensciencegrid_org && has_avx && has_avx2",
+        "environment": f'"{environment_str}"',  # must be quoted
+        "+FileSystemDomain": '"blah"',  # must be quoted
+        #
+        "should_transfer_files": "YES",
+        "transfer_input_files": f'"{input_files_str}"',  # must be quoted
+        "transfer_output_files": '""',  # must be quoted for "none"
+        #
+        # Don't transfer executable (/bin/bash) in case of
+        #   version (dependency) mismatch.
+        #     Ex:
+        #     "/lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.36' not found"
+        # Technically this is just needed for spooling -- since if
+        #   we don't spool, the executable (/bin/bash) can't be
+        #   transferred anyway and so a local version will be used
+        "transfer_executable": "false",
+        #
+        "request_cpus": str(n_cores),
+        "request_memory": humanfriendly.format_size(  # 1073741824 -> "1 GiB" -> "1 GB"
+            worker_memory_bytes, binary=True
+        ).replace("i", ""),
+        "request_disk": humanfriendly.format_size(  # 1073741824 -> "1 GiB" -> "1 GB"
+            worker_disk_bytes, binary=True
+        ).replace("i", ""),
+        "priority": int(priority),
+        "+WantIOProxy": "true",  # for HTChirp
+        "+OriginalTime": max_worker_runtime,  # Execution time limit -- 1 hour default on OSG
+        #
+        "+EWMSTaskforceUUID": f'"{taskforce_uuid}"',  # must be quoted
+        "job_ad_information_attrs": "EWMSTaskforceUUID",
+    }
+
+    # outputs
+    if do_transfer_output:
+        # this is the location where the files will go when/if *returned here*
+        logs_dir = make_condor_logs_dir()
+        submit_dict.update(
+            {
+                "output": str(logs_dir / "tms-worker-$(ProcId).out"),
+                "error": str(logs_dir / "tms-worker-$(ProcId).err"),
```

Comment:
Might want to also include the `ClusterId`, since there will be multiple clusters per logdir.

Reply:
got it :+1: